src/roiextractors/memmapextractors.py

```diff
@@ -2,109 +2,118 @@
-        save_path = Path(self._tmp_file)
-        self._video = np.memmap(
-            save_path,
-            shape=(
-                self.imaging.get_num_channels(),
-                self.imaging.get_num_frames(),
-                self.imaging.get_image_size()[0],
-                self.imaging.get_image_size()[1],
-            ),
-            dtype=self.imaging.get_dtype(),
-            mode="w+",
-        )
-
-        if verbose:
-            for ch in range(self.imaging.get_num_channels()):
-                print(f"Saving channel {ch}")
-                for i in tqdm(range(self.imaging.get_num_frames())):
-                    plane = self.imaging.get_frames(i, channel=ch)
-                    self._video[ch, i] = plane
-        else:
-            for ch in range(self.imaging.get_num_channels()):
-                for i in range(self.imaging.get_num_frames()):
-                    plane = self.imaging.get_frames(i, channel=ch)
-                    self._video[ch, i] = plane
-
-    @property
-    def filename(self):
-        return str(self._tmp_file)
-
-    def move_to(self, save_path):
-        save_path = Path(save_path)
-        if save_path.suffix != ".dat" and save_path.suffix != ".bin":
-            save_path = save_path.with_suffix(".dat")
-        if not save_path.parent.is_dir():
-            os.makedirs(save_path.parent)
-        shutil.move(self._tmp_file, str(save_path))
-        self._tmp_file = str(save_path)
-        self._video = np.memmap(
-            save_path,
-            shape=(
-                self.imaging.get_num_channels(),
-                self.imaging.get_num_frames(),
-                self.imaging.get_image_size()[0],
-                self.imaging.get_image_size()[0],
-            ),
-            dtype=self.imaging.get_dtype(),
-            mode="r",
-        )
-
-    @check_get_frames_args
-    def get_frames(self, frame_idxs, channel=0):
-        return self._video[channel, frame_idxs]
+        """Class for reading binary data.
+
+
+        Parameters
+        ----------
+        file_path : PathType
+            the file_path where the data resides.
+        frame_shape : tuple
+            The frame shape of the image determines how each frame looks. Examples:
+            (n_channels, rows, columns), (rows, columns, n_channels), (n_channels, columns, rows), etc.
+            Note that n_channels is 1 for grayscale and 3 for color images.
+        dtype : DtypeType
+            The type of the data to be loaded (int, float, etc.)
+        offset : int, optional
+            The offset in bytes. Usually corresponds to the number of bytes occupied by the header. 0 by default.
+        sampling_frequency : float, optional
+            The sampling frequency.
+        image_structure_to_axis : dict, optional
+            A dictionary indicating what axis corresponds to what in the memmap. The default values are:
+            dict(frame_axis=0, num_channels=1, rows=2, columns=3)
```

Comment:
Are 'rows' and 'columns' the best verbal way to communicate the (x,y) dimensions of an image? Must think on this...

Reply:
Yes, some libraries have (x, y) but that's ambiguous as sometimes this intuition crashes with our picture of the Cartesian plane. Some areas have x means rows and others have y means rows:

```
 ^
 |
 |
 Y
 |
 |
0/0---X--->
````